### ISSUE TITLE
Bad Request response for api/cla/getAll

### ISSUE THREAD

**Issue by mcred** [cla-assistant/cla-assistant#331]
Based on #166, I created a request to `https://cla-assistant.io/api/cla/getAll` and set the `x-token` header to be an access token that has access to my org. I've confirmed that the CLA Assistant app has proper permissions, as well as organization access. When I pass the following as the request body:

```
{
	"orgId":"682240",
	"gist":{
		"gist_url":"https://gist.github.com/a072029a2c59d6d25e774921fa1d88d1"
		
	}
	
}
```
I receive the following response:
```
{
    "message": {
        "code": "400",
        "status": "Bad Request",
        "message": "Empty value for parameter 'id': undefined"
    }
}
```
Can someone help me understand why the request is bad and how to fix it?

**Comment by KharitonOff** [comment]
indeed, it doesn't work currently. This API should still work for repos but it's broken for orgs. Sorry for that. I'll fix it in the next release. It could take a one or two weeks, though. I hope you don't depend on it.

**Comment by mcred** [comment]
In the changelog for 1.7.0 I see this issue reference. I tried my call again, but now I get the following response:

```
{
    "message": "You have no push permission for this org or repo"
}
```

Which org or repo is this message referring to? For the org in question, I'm an owner and have admin access to all repos.

**Comment by KharitonOff** [comment]
@mcred yes, you are right. It seems that you have linked your org with a 'shared' CLA. The signatures for this CLA (can) refer to multiple orgs and repos. So this API doesn't work for shared CLAs yet. Sorry for that! Unfortunately I can't make any forecast how soon we'll make it possible to get the signatures via the API for a shared CLA.

**Comment by mcred** [comment]
Do you have any suggestions for a work around? It works through the UI. See screenshots below. Is there a way to make multiple API calls or act like the UI? 

<img width="858" alt="screen shot 2018-06-14 at 10 05 35 am" src="https://user-images.githubusercontent.com/4326032/41418048-efcd6562-6fbc-11e8-8974-ae4a1a8b4430.png">
<img width="858" alt="screen shot 2018-06-14 at 10 07 47 am" src="https://user-images.githubusercontent.com/4326032/41418049-eff13a28-6fbc-11e8-8bb0-711c224a32de.png">

**Comment by KharitonOff** [comment]
@mcred sorry for late response. Unfortunately I don't have a quick solution in mind. I see the need of improvement for the API, but I can not estimate how soon we'll make it available. The fastest way probably would be a corresponding contribution from your side 😃 Otherwise I'll keep in mind your request and will implement a solution after the features of higher priority are done.

**Comment by tjrgg** [comment]
Is anyone working on this?

**Comment by ibakshay** [comment]
Hi @tjrgg, we are not working on this currently as we are focusing on more important topics. Feel free to raise a Pull Request If you have time and capacity to do so 😄
